iphone_connector.py

```diff
@@ -119,11 +119,33 @@ def initialize():
     # of the files within them.
     has_subdirectory_structure = MESSAGE_DB[:2] in os.listdir(_latest_sync_dir)
     if has_subdirectory_structure:
-        _message_con = sqlite3.connect(os.path.join(_latest_sync_dir, MESSAGE_DB[:2], MESSAGE_DB))
-        _address_con = sqlite3.connect(os.path.join(_latest_sync_dir, ADDRESS_DB[:2], ADDRESS_DB))
+        message_path = os.path.join(_latest_sync_dir, MESSAGE_DB[:2], MESSAGE_DB)
+        address_path = os.path.join(_latest_sync_dir, ADDRESS_DB[:2], ADDRESS_DB)
     else:
-        _message_con = sqlite3.connect(os.path.join(_latest_sync_dir, MESSAGE_DB))
-        _address_con = sqlite3.connect(os.path.join(_latest_sync_dir, ADDRESS_DB))
+        message_path = os.path.join(_latest_sync_dir, MESSAGE_DB)
+        address_path = os.path.join(_latest_sync_dir, ADDRESS_DB)
+
+    _message_con = sqlite3.connect(message_path)
+    _address_con = sqlite3.connect(address_path)
+
+    # Try to read from the DB files as a check if they are encrypted and raise exception accordingly
+    try:
+        t = ('table',)
+        _message_con.execute("SELECT name FROM SQLITE_MASTER where type=?", t)
+    except sqlite3.DatabaseError:
+        raise sqlite3.DatabaseError(
+            "A sqlite connection to the file at {0} failed, perhaps you've set iTunes to use encrypted backup?"
+            .format(message_path)
+        )
+    
+    try:
```

Comment:
This section seems unnecessary, I believe encryption is all or nothing, so it suffices to check just one of the tables.  Add a comment in the code stating this.